src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/JintLuceneDocumentConverter.cs

```diff
@@ -346,32 +362,47 @@ private static JsValue TryDetectDynamicFieldCreation(string property, ObjectInst
                     var propertyNameAsString = propertyName.AsString();
                     if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.Indexing), StringComparison.OrdinalIgnoreCase))
                     {
-                        field.Indexing = GetEnum<FieldIndexing>(optionValue, propertyNameAsString);
+                        fieldIndexing = GetEnum<FieldIndexing>(optionValue, propertyNameAsString);
 
                         continue;
                     }
 
                     if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.Storage), StringComparison.OrdinalIgnoreCase))
                     {
                         if (optionValue.IsBoolean())
-                            field.Storage = optionValue.AsBoolean()
+                            fieldStorage = optionValue.AsBoolean()
                                 ? FieldStorage.Yes
                                 : FieldStorage.No;
                         else
-                            field.Storage = GetEnum<FieldStorage>(optionValue, propertyNameAsString);
+                            fieldStorage = GetEnum<FieldStorage>(optionValue, propertyNameAsString);
 
                         continue;
                     }
 
                     if (string.Equals(propertyNameAsString, nameof(CreateFieldOptions.TermVector), StringComparison.OrdinalIgnoreCase))
                     {
-                        field.TermVector = GetEnum<FieldTermVector>(optionValue, propertyNameAsString);
+                        termVector = GetEnum<FieldTermVector>(optionValue, propertyNameAsString);
 
                         continue;
                     }
                 }
             }
 
+            field = IndexField.Create(fieldName, 
```

Comment:
do we need to clone the `field` or creating a new one is enough?

Reply:
Field here is local ref variable, we're changing reference to new object. Object inside `IndexDefiniton` is not changed by current flow.